apstools/utils.py

```diff
@@ -384,7 +393,377 @@ def itemizer(fmt, items):
+
+    EXAMPLE::
+
+        ListRuns(cat).to_dataframe()
+
+    PUBLIC METHODS
+
+    .. autosummary::
+
+        ~to_dataframe
+        ~to_table
+        ~parse_runs
+
+    INTERNAL METHODS
+
+    .. autosummary::
+
+        ~_get_by_key
+        ~_check_cat
+        ~_apply_search_filters
+        ~_sorter
+        ~_check_keys
+
+    """
+
+    cat: object = None
+    query: dict = None
+    keys: str = None
+    missing: str = ""
+    num: int = 20
+    reverse: bool = True
+    since: str = None
+    sortby: str = "time"
+    timefmt: str = "%Y-%m-%d %H:%M:%S"
+    until: str = None
+
+    _default_keys = "scan_id time plan_name detectors"
+
+    def _get_by_key(self, md, key):
+        """
+        Get run's metadata value by key.
+
+        Look in ``start`` document first.
+        If not found, look in ``stop`` document.
+        If not found, report using ``self.missing``.
+
+        If ``key`` is found but value is None, report as ``self.missing``.
+
+        The ``time`` key will be formatted by the ``self.timefmt`` value.
+        See https://strftime.org/ for examples.  The special ``timefmt="raw"``
+        is used to report time as the raw value (floating point time as used in
+        python's ``time.time()``).
+
+        A special syntax of ``key`` allows reporting of keys in other
+        metadata subdictionaries.  The syntax is ``doc.key`` (such as
+        specifying the ``time`` key  from the ``stop`` document:
+        ``stop.time``) where a ``.`` is used to separate the
+        subdictionary name (``doc``) from the ``key``.
+        **Note**:
+        It is not possible to ``sortby`` the dotted-key syntax
+        at this time.
+        """
+        v = None
+        if key == "time":
+            v = md["start"][key]
+            if self.timefmt != "raw":
+                ts = datetime.datetime.fromtimestamp(v)
+                v = ts.strftime(self.timefmt)
+        elif key in md["start"]:
+            v = md["start"].get(key, self.missing)
+        elif md["stop"] and key in md["stop"]:
+            v = md["stop"].get(key, self.missing)
+        elif len(key.split(".")) == 2:
+            # dotted-key syntax
+            doc, key = key.split(".")
+            if md[doc] is not None:
+                v = md[doc].get(key, self.missing)
+                if key == "time" and self.timefmt != "raw":
+                    ts = datetime.datetime.fromtimestamp(v)
+                    v = ts.strftime(self.timefmt)
+        return v or self.missing
+
+    def _check_cat(self):
+        if self.cat is None:
+            self.cat = getCatalog()
+
+    def _apply_search_filters(self):
+        since = self.since or "1995-01-01"
+        until = self.until or "2100-12-31"
+        self._check_cat()
+        cat = self.cat.v2.search(
+            databroker.queries.TimeRange(since=since, until=until)
+        ).search(self.query or {})
+        return cat
+
+    def parse_runs(self):
```

Comment:
It will be very useful to have the option to get the metadata for particular runs instead of the last few. So should we add an option here to ask for specific runs (by `uid` or `scan_id`)?

Reply:
Since I want to release in the next few days and this could take a bit longer to implement well, I'm moving this comment to be a new issue for the next release.